addons/website/models/website.py

```diff
@@ -288,9 +288,19 @@ def configurator_skip(self):
     @api.model
     def configurator_apply(self, **kwargs):
         def set_colors(selected_palette):
-            url = '/website/static/src/scss/options/user_values.scss'
-            custo = {'color-palettes-name': "'%s'" % selected_palette}
-            self.env['web_editor.assets'].make_scss_customization(url, custo)
+            if selected_palette.get('is_recommended'):
+                url = '/website/static/src/scss/options/colors/user_color_palette.scss'
+                values = {}
+                for k, v in selected_palette.items():
+                    color_match = re.match('^color([1-5])$', k)
+                    if color_match:
+                        color_key = 'o-color-' + color_match.group(1)
+                        values[color_key] = v
+            else:
+                url = '/website/static/src/scss/options/user_values.scss'
+                values = {'color-palettes-name': "'%s'" % selected_palette['name']}
+            WebEditorAssets = self.env['web_editor.assets'].with_context(install_module=[kwargs['theme_name']])
```

Comment:
I had to use this with_context otherwise it fails... The new theme is installed before making the scss customization but it doesn't seem to be in the registry's _init_modules

![Screenshot from 2021-04-19 14-05-18](https://user-images.githubusercontent.com/49784121/115356165-aab6e000-a1bb-11eb-87eb-7ee122d50259.png)
![Screenshot from 2021-04-19 14-02-44](https://user-images.githubusercontent.com/49784121/115356177-ad193a00-a1bb-11eb-9e2d-8ede06b28936.png)
![Screenshot from 2021-04-19 14-02-50](https://user-images.githubusercontent.com/49784121/115356172-abe80d00-a1bb-11eb-9913-620848f20a45.png)

Reply:
I don't understand what you mean here. Where is the notion of theme and/or installation when you call this method?

Reply:
> I don't understand what you mean here. Where is the notion of theme and/or installation when you call this method?

If you look at the traceback from the call to `make_css_customizations` you reach a call to `_get_paths` which raise an error because the addon (`theme_bistro`)  is not in the list of installed theme. But `theme_bistro` is installed before we make the call to `make_css_customization` in `configurator_apply`.
![Screenshot from 2021-04-23 09-31-24](https://user-images.githubusercontent.com/49784121/115836477-d209e900-a417-11eb-958f-d88b1c543c1b.png)
The list of installed addons comes from
![Screenshot from 2021-04-23 09-46-32](https://user-images.githubusercontent.com/49784121/115837556-016d2580-a419-11eb-8630-c0598a8b54bc.png)
`theme_bistro` is not in `self.env.registry._init_modules` then what I found as solution was to add it to the context.

Reply:
Mmh, ok, at least add a comment to explain that in the code :wink:

Reply:
After discussion with @odony , the problem is that you cannot use the environment after an installation of module.
So in you case, after install theme, or module, you need to use a new environment.

```py
theme.button_choose_theme()

# Force to refresh env after install of module
request._cr.commit()
api.Environment.reset()
request._env =  api.Environment(theme._cr, theme._uid, theme._context)

```

Now you have your module in env.registry._init_modules as expected